pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -255,67 +279,133 @@ func (p *PoolManager) initVirtualMachineMap() error {
 	}
 
 	for _, vm := range vms.Items {
-		logger.V(1).Info("InitMaps for virtual machine",
-			"virtualMachineName", vm.Name,
-			"virtualMachineNamespace", vm.Namespace)
-		if len(vm.Spec.Template.Spec.Domain.Devices.Interfaces) == 0 {
+		vmFullName := VmNamespaced(&vm)
+		vmInterfaces := getVirtualMachineInterfaces(&vm)
+		vmNetworks := getVirtualMachineNetworks(&vm)
+		logger.V(1).Info("InitMaps for virtual machine")
+		if len(vmInterfaces) == 0 {
 			logger.V(1).Info("no interfaces found for virtual machine, skipping mac allocation", "virtualMachine", vm)
 			continue
 		}
 
-		if len(vm.Spec.Template.Spec.Networks) == 0 {
+		if len(vmNetworks) == 0 {
 			logger.V(1).Info("no networks found for virtual machine, skipping mac allocation",
 				"virtualMachineName", vm.Name,
 				"virtualMachineNamespace", vm.Namespace)
 			continue
 		}
 
 		networks := map[string]kubevirt.Network{}
-		for _, network := range vm.Spec.Template.Spec.Networks {
+		for _, network := range vmNetworks {
 			networks[network.Name] = network
 		}
 
 		logger.V(1).Info("virtual machine data",
-			"virtualMachineName", vm.Name,
-			"virtualMachineNamespace", vm.Namespace,
-			"virtualMachineInterfaces", vm.Spec.Template.Spec.Domain.Devices.Interfaces)
+			"vmFullName", vmFullName,
+			"virtualMachineInterfaces", vmInterfaces)
 
-		for _, iface := range vm.Spec.Template.Spec.Domain.Devices.Interfaces {
+		for _, iface := range vmInterfaces {
 			if iface.Masquerade == nil && iface.Slirp == nil && networks[iface.Name].Multus == nil {
 				logger.Info("mac address can be set only for interface of type masquerade and slirp on the pod network")
 				continue
 			}
```

Comment:
I'll move it to the inner functions

>(btw incase of interface without network this will blow up)

From what I experimented then you can't create a vm with network without network...